src/NamespaceRepository.php

```diff
@@ -39,6 +44,11 @@ class NamespaceRepository {
 	public function __construct() {
 		$this->canonical_namespaces = [ NS_MAIN => 'Main' ] + $this->getConfig()->get( 'CanonicalNamespaceNames' );
 		$this->extension_namespaces = ExtensionRegistry::getInstance()->getAttribute( 'ExtensionNamespaces' );
+		$this->dbLoadBalancer = self::getDBLoadBalancer();
+	}
+
+	private static function getDBLoadBalancer() {
+		return MediaWikiServices::getInstance()->getDBLoadBalancer();
```

Comment:
I wouldn't wrap this call into a static function. It's short enough to be repeated in three places it is used.